shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/service/DatasetServiceImpl.java

```diff
@@ -394,6 +396,10 @@ public List<Dataset> findByExaminationId(Long examinationId) {
 		return Utils.toList(repository.findByDatasetAcquisitionExaminationId(examinationId));
 	}
 
+	public List<Dataset> findDatasetAndOutputByExaminationId(Long examinationId) {
+        return StreamSupport.stream(repository.findAllById(repository.findDatasetAndOutputByExaminationId(examinationId)).spliterator(), false).toList();
```

Comment:
Hi @DuckflipXYZ, thank you for your PR.
This looks like a very database intensive pile of code and I am not sure I understand, what it does.
With kind regards, Michael